### ISSUE TITLE
Display a warning if incompatible modules/plugins are active

### ISSUE THREAD

**Issue by Zodiac1978** [pluginkollektiv/antispam-bee#64]
We know that ASB is not compatible with Jetpack Comments module and some other Ajax powered comment plugins. We can collect them and display a warning if they are activated.

**Comment by Zodiac1978** [comment]
```
if ( class_exists( 'Jetpack' ) && Jetpack::is_module_active( 'comments' ) ) :
endif;
```
See: http://wptheming.com/2013/04/check-if-jetpack-modules-are-enabled/

```
if( is_plugin_active( 'disqus-comment-system/disqus.php' ) ) {
	// EDD is active
}
```
See: https://pippinsplugins.com/checking-dependent-plugin-active/

Alternative:
```if (defined('DISQUS_VERSION')) { ... }```

**Comment by Zodiac1978** [comment]
Maybe we can just skip the honeypot feature for Jetpack and leave the rest? Jetpack is using their own comment form, but it is saving to WP, so our other features should work, or not?

**Comment by Zodiac1978** [comment]
Adding "Instant Comment Validation" to the virtual list of incompatible plugins.
See: https://wordpress.org/support/topic/all-comments-are-rejected-v-2-7-0/#post-8882790

**Comment by Zodiac1978** [comment]
ThriveThemes (per support forum thread)
https://wordpress.org/support/topic/feature-request-disable-honey-pot-switch/

Maybe we can disable the honeypot in this case and provide all other checks in those cases, like suggested in this thread?

**Comment by websupporter** [comment]
I move this issue to "Future Release" milestone. I hesitate to say "lets put this feature in the plugin" as I think its quite some overhead to maintain. Missing plugins might lead to "but you didnt list the plugin, so you should keep care of ASB works with it", something like this. Plugins might become compatible in the future due to changes in those plugins, so the list needs to be regularly checked and compatible plugins might need to be removed.

ThriveThemes as an example is a premium product, to really properly maintain ThriveThemes as "not compatible", we would need this theme.

That said, maybe a Wiki page would be sufficient already for such a list. I do not rule against "Display a warning", but I would like to see where to place this warning. Some mockup would be nice.

**Comment by Zodiac1978** [comment]
What about making the honeypot feature an option? The other checks may still work (except for complete comment replacements like Disquz). I still think this is a feature which could help us, but at best with a solution that don't puts any maintaining pressure on us.

**Comment by websupporter** [comment]
To make it optional, I would prefer to reconsider this for after ASB 3.0, as this means some refactoring of the plugin.

**Comment by Zodiac1978** [comment]
> Missing plugins might lead to "but you didnt list the plugin, so you should keep care of ASB works with it", something like this.

I don't think this is the case, because you just see the warning if an incompatible plugin is active. The same argument can be said about our FAQ at this moment. We could limit this feature to the big ones (mainly Jetpack Comments and other like wpdiscus or Disqus) which will never work with our RegEx for the honeypot.

> Some mockup would be nice.

I would just use the default admin notice as a warning (orange) or error (red).
